File: 11.1P - Clock in Another Language/Counter.cpp
#include <string>

using namespace std;

class Counter {
private:
    int _count;
    string _name;

public:
    Counter(const string& name) : _count(0), _name(name) {}

    void Increment() {
        ++_count;
    }

    void Reset() {
        _count = 0;
    }

    string Name() const {
        return _name;
    }

    void SetName(const string& value) {
        _name = value;
    }

    int Ticks() const {
        return _count;
    }
};

File: 3.1P Clock Class/ClockApplication/Counter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClockApplication
{
    public class Counter
    {
        private int _count;

        public Counter()
        {
            _count = 0;
        }

        public void Increment()
        {
            _count++;
        }

        public void Reset()
        {
            _count = 0;
        }

        public int Tick()
        {
            return _count;
        }
    }
}
